### PR TITLE
Adding Cloudformation template, bash automation and documentation update

### PR DESCRIPTION
**Pull request by aljazceru**

### CODE REVIEW

File: docs/CLN.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+1. Define the environment variables as described in [sample.env](./sample.env). If `CERTMAGIC_DOMAIN` is defined, certificate for this domain is automatically obtained and renewed from Let's Encrypt. In this case, the port needs to be 443. If `CERTMAGIC_DOMAIN` is not defined, lspd needs to run behind a reverse proxy like treafik or nginx.
+
+ENV variables:
+- `LISTEN_ADDRESS` defines the host:port for the lspd grpc server
+- `CERTMAGIC_DOMAIN` domain on which lspd will be accessible
+- `DATABASE_URL` postgresql db url
+- `AWS_REGION` AWS region for SES emailing 
+- `AWS_ACCESS_KEY_ID` API key for SES emailing 
+- `AWS_SECRET_ACCESS_KEY`API secret  for SES emailing 
+- `MEMPOOL_API_BASE_URL` uses fee estimation for opening new channels (default: https://mempool.space)
```
Comment (JssDWt) [inline docs/CLN.md:50]:
```suggestion
- `MEMPOOL_API_BASE_URL` uses fee estimation for opening new channels (default: https://mempool.space/api/v1/)
```

File: docs/CLN.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+1. Create a random token (for instance using the command `openssl rand -base64 48`, or `./lspd genkey`)
+1. Define the environment variables as described in [sample.env](./sample.env). If `CERTMAGIC_DOMAIN` is defined, certificate for this domain is automatically obtained and renewed from Let's Encrypt. In this case, the port needs to be 443. If `CERTMAGIC_DOMAIN` is not defined, lspd needs to run behind a reverse proxy like treafik or nginx.
+
+ENV variables:
+- `LISTEN_ADDRESS` defines the host:port for the lspd grpc server
+- `CERTMAGIC_DOMAIN` domain on which lspd will be accessible
+- `DATABASE_URL` postgresql db url
+- `AWS_REGION` AWS region for SES emailing 
+- `AWS_ACCESS_KEY_ID` API key for SES emailing 
+- `AWS_SECRET_ACCESS_KEY`API secret  for SES emailing
```
Comment (JssDWt) [inline docs/CLN.md:49]:
```suggestion
- `AWS_SECRET_ACCESS_KEY`API secret for SES emailing 
```

File: docs/CLN.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+ENV variables:
+- `LISTEN_ADDRESS` defines the host:port for the lspd grpc server
+- `CERTMAGIC_DOMAIN` domain on which lspd will be accessible
+- `DATABASE_URL` postgresql db url
+- `AWS_REGION` AWS region for SES emailing 
+- `AWS_ACCESS_KEY_ID` API key for SES emailing 
+- `AWS_SECRET_ACCESS_KEY`API secret  for SES emailing 
+- `MEMPOOL_API_BASE_URL` uses fee estimation for opening new channels (default: https://mempool.space)
+- `MEMPOOL_PRIORITY` priority with which open new channels using mempool api (default: economy)
```
Comment (JssDWt) [inline docs/CLN.md:51]:
```suggestion
- `MEMPOOL_PRIORITY` priority with which open new channels using mempool api 
  (options: minimum, economy, hour, halfhour, fastest) (default: economy)
```

File: docs/CLN.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+In order to run lspd on top of CLN, you need to run the lspd process and run cln with the provided cln plugin. You also need lightningd compiled with developer mode on (`./configure --enable-developer`)
+
+The cln plugin (go build -o lspd_plugin cln_plugin/cmd) is best started with a bash script to pass environment variables (note this LISTEN_ADDRESS is the listen address for communication between lspd and the plugin, this is not the listen address mentioned in the 'final step')
+
+```bash
+#!/bin/bash
+
+export LISTEN_ADDRESS=<listen address>
+/path/to/lspd_plugin
+```
```
Comment (JssDWt) [inline docs/CLN.md:65]: This startup of the plugin is out-of-date (I see it's out-of-date in the main readme too, sorry)

The cln plugin exposes a grpc interface that lspd uses to intercept htlcs.
The cln plugin is started by passing commandline parameters to cln, like so.
`lightningd --plugin=/path/to/lspd_plugin --lsp-listen=<grpc listen address>`

Optional:
- `--lsp-subscribertimeout=1m`: If lspd is not connected to the plugin, htlcs will 'hang'. If lspd hasn't connected within the subscriber timeout, the htlc is continued as if no lsp exists. Valid values are golang duration strings. (default: 1m)
- `--lsp-channel-accept-script`: Optional channel acceptor script for inbound channels. It's a starlark script that receives the `openchannel` or `openchannel2` request from cln, and needs to return a valid response for the hook. Example: `--lsp-channel-accept-script=(lambda oc: encode({"result":"continue"}) if oc["id"]=="031015a7839468a3c266d662d5bb21ea4cea24226936e2864a7ca4f2c3939836e0" else encode({"result":"reject", "error_message": "The channel cannot be opened"}))(decode(openchannel))`

File: docs/LND.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+#### LND
+Follow LND installation instructions [here](https://github.com/lightningnetwork/lnd/blob/master/docs/INSTALL.md). 
+
+#### lspd
+Needs to be build from source:
+```
+git clone https://github.com/breez/lspd 
+cd lspd
+go build .                      # compile lspd
+go build -o lspd_plugin         # compile lspd cln plugin
```
Comment (JssDWt) [inline docs/LND.md:18]: This line can be removed
```suggestion
```

File: docs/LND.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+```
+### Postgresql
+Lspd supports postgresql backend. Create new database and user for lspd:
+```
+CREATE ROLE <username>;
+ALTER ROLE <username> WITH NOSUPERUSER INHERIT NOCREATEROLE NOCREATEDB LOGIN NOREPLICATION NOBYPASSRLS PASSWORD '<password>';
+CREATE DATABASE <dbname> WITH TEMPLATE = template0 ENCODING = 'UTF8' LC_COLLATE = 'en_US.UTF-8' LC_CTYPE = 'en_US.UTF-8';
+ALTER DATABASE <dbname> OWNER TO <username>;
+``````
+
```
Comment (JssDWt) [inline docs/LND.md:28]: CLN has another section specifically for AWS here.

File: docs/aws.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+Cloudformation template for automated deployment of lspd, bitcoind and cln with postgresql backend.
+### Requirements 
+- AWS account 
+- AWS SES configured
+
+### Deployment 
+[Cloudformation template](../deploy/deploy.yml) will automatically deploy several things:
+- new ec2 instance (m6a.xlarge) to your selected VPC
+- bitcoind
+- clnd (with postgresql as backend)
```
Comment (JssDWt) [inline docs/aws.md:11]: this made me laugh out loud

File: deploy/deploy.yml [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+            git checkout v23.05
+            ./configure --enable-developer
+            make
+            make install
+            cat <<EOL | sudo tee /etc/systemd/system/lightningd.service
+            [Unit]
+            Description=Lightning Network Daemon (lightningd)
+            After=network.target
+            [Service]
+            ExecStart=/usr/local/bin/lightningd --plugin=/home/lightning/.lightning/plugins/lspd_plugin  --lsp-listen=127.0.0.1:12312 --max-concurrent-htlcs=30
```
Comment (JssDWt) [inline deploy/deploy.yml:207]: This is missing `--dev-allowdustreserve=true` and `--allow-deprecated-apis=true`

File: deploy/deploy.yml [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+            sudo env "PATH=$PATH" go get github.com/breez/lspd/cln_plugin
+            sudo env "PATH=$PATH" go build .
+            sudo env "PATH=$PATH" go build -o lspd_plugin ./cln_plugin/cmd
+            sudo cp lspd /usr/local/bin/
+            sudo mkdir /home/lightning/.lightning/plugins
+            sudo cp lspd_plugin /home/lightning/.lightning/plugins/
+
+            cat <<EOL | sudo tee /etc/systemd/system/lspd.service
+            [Unit]
+            Description=Lightning Service Daemon (lspd)
```
Comment (JssDWt) [inline deploy/deploy.yml:233]:
```suggestion
            Description=Lightning Service Provider Daemon (lspd)
```

File: deploy/lspd-install.sh [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+git checkout v23.05
+./configure --enable-developer
+make
+make install
+cat <<EOL | sudo tee /etc/systemd/system/lightningd.service
+[Unit]
+Description=Lightning Network Daemon (lightningd)
+After=network.target
+[Service]
+ExecStart=/usr/local/bin/lightningd --plugin=/home/lightning/.lightning/plugins/lspd_plugin  --lsp-listen=127.0.0.1:12312 --max-concurrent-htlcs=30
```
Comment (JssDWt) [inline deploy/lspd-install.sh:171]: This is missing `--dev-allowdustreserve=true` and `--allow-deprecated-apis=true`

File: docs/CLN.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+1. Run cln with the following options set:
+    - `--plugin=/path/to/shell/script.sh`: to use lspd as plugin
+    - `--max-concurrent-htlcs=30`: In order to use zero reserve channels on the client side, (local max_accepted_htlcs + remote max_accepted_htlcs + 2) * dust limit must be lower than the channel capacity. Reduce max-concurrent-htlcs or increase channel capacity accordingly.
+    - `--dev-allowdustreserve=true`: In order to allow zero reserve on the client side (requires developer mode turned on)
```
Comment (JssDWt) [inline docs/CLN.md:70]: Also
- `--allow-deprecated-apis=true`: lspd currently uses a deprecated api, so needs this flag set.

File: README.md [hunk 12]
```diff
@@ -3,6 +3,77 @@ lspd is a simple deamon that provides [LSP](https://medium.com/breez-technology/
...
+- lnd process
+- bitcoind process
+- docker container for postgres with default name
+
+It may be a good idea to clean your testdir every once in a while if you're 
+using the `preservel# lspd simple server
+lspd is a simple deamon that provides [LSP](https://medium.com/breez-technology/introducing-lightning-service-providers-fe9fb1665d5f) services to [Breez clients](https://github.com/breez/breezmobile).   
+
+This is a simple example of an lspd that works with an [lnd](https://github.com/lightningnetwork/lnd) node or a [cln](https://github.com/ElementsProject/lightning) node.
+
```
Comment (JssDWt) [inline README.md:76]: There's a large duplicate chunk below

**Review by JssDWt — CHANGES_REQUESTED** [review]
It looks very promising! I'm excited.
I went over all the docs, have a few nits.

File: docs/CLN.md (hunk 11 above)
Comment (JssDWt) [reply docs/CLN.md:66]: This still applies

**Comment by JssDWt** [comment]
Also this one
https://github.com/breez/lspd/pull/119#discussion_r1305624870

File: docs/CLN.md (hunk 11 above)
Comment (aljazceru) [reply docs/CLN.md:66]: oh yeah, it was added to installer but not documentation

**Review by yaslama — APPROVED** [review]

**Pull request merged**